test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs

```diff
@@ -1297,6 +1297,56 @@ public void ControllerView_UsesControllerViewEngine()
             Assert.Same(viewEngine, result.ViewEngine);
         }
 
+        [Fact]
+        public void TryValidateModelWithValidModel_ReturnsTrue()
+        {
+            // Arrange
+            var binder = new Mock<IModelBinder>();
+            var controller = GetController(binder.Object, provider: null);
+            controller.BindingContext.ValidatorProvider = Mock.Of<IModelValidatorProvider>();
+            var model = new TryValidateModelModel();
+
+            // Act
+            var result = controller.TryValidateModel(model);
+
+            // Assert
+            Assert.True(result);
+            Assert.True(controller.ModelState.IsValid);
+        }
+
+        [Fact]
+        public void TryValidateModelWithInvalidModel_ReturnsFalse()
+        {
+            // Arrange
+            var model = new TryValidateModelModel();
+            var validationResult = new []
+                 {
+                    new ModelValidationResult(string.Empty, "Out of range!")
+                 };
+
+            var validator1 = new Mock<IModelValidator>();
+
+            validator1.Setup(v => v.Validate(It.IsAny<ModelValidationContext>()))
+               .Returns(validationResult);
+
+            var provider = new Mock<IModelValidatorProvider>();
+            provider.Setup(v => v.GetValidators(It.IsAny<ModelMetadata>()))
+                .Returns(new[] { validator1.Object });
+
+            var binder = new Mock<IModelBinder>();
+            var controller = GetController(binder.Object, provider: null);
+            controller.BindingContext.ValidatorProvider = provider.Object;
+            
+            // Act
+            var result = controller.TryValidateModel(model, "Prefix");
```

Comment:
Can we have a test without the prefix to ensure the modelstate keys are correctly generated?